spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/ColumnType.scala

```diff
@@ -84,7 +92,8 @@ object ColumnType {
 
   def fromDriverType(dataType: DataType): ColumnType[_] = {
     val typeArgs = dataType.getTypeArguments.map(fromDriverType)
-    (dataType, dataType.getName) match {
+
+    val standardFromDriverRow: PartialFunction[(DataType, DataType.Name), ColumnType[_]] = {
```

Comment:
More or less, I was thinking about simplifying this a little bit:

```scala
  private def typeArg(dataType: DataType, idx: Int) = fromDriverType(dataType.getTypeArguments.get(idx))

  private val standardFromDriverRow: PartialFunction[DataType, ColumnType[_]] = {
    case listType if listType.getName == DataType.Name.LIST => ListType(typeArg(listType, 0))
    case setType if setType.getName == DataType.Name.SET => SetType(typeArg(setType, 0))
    case mapType if mapType.getName == DataType.Name.MAP => MapType(typeArg(mapType, 0), typeArg(mapType, 1))
    case (userType: DriverUserType, _) => UserDefinedType(userType.getTypeName, fields(userType))
    case (tupleType: DriverTupleType, _) => TupleType(fields(tupleType): _*)
    case (dataType) => primitiveTypeMap(dataType)
  }


  def fromDriverType(dataType: DataType): ColumnType[_] = {
    val getColumnType: PartialFunction[DataType, ColumnType[_]] = customFromDriverRow orElse standardFromDriverRow
    getColumnType((dataType, dataType.getName))
  }
```

In particular, why this partial function needs to take both `DataType` and value of `name` property of it. As long as it was somewhere deep in connector code as a super private piece of code it was ok. But now we are going to have public interface which would be a little odd if it had redundant parameters. wdyt ?